patch/misc/general-packaging-5.3.y.patch

```diff
@@ -180,36 +209,47 @@ index 5858d6e44..e81b8a6fc 100644
 -	create_package "$kernel_headers_packagename" "$kernel_headers_dir"
 -	create_package "$libc_headers_packagename" "$libc_headers_dir"
 +	create_package "$kernel_headers_packagename" "$kernel_headers_dir" "headers"
++#UNCOMENT	create_package "$libc_headers_packagename" "$libc_headers_dir"
 +	create_package "$dtb_packagename" "$dtb_dir" "dtb"
  fi
  
  create_package "$packagename" "$tmpdir"
+diff --git a/scripts/package/mkdebian b/scripts/package/mkdebian
+index e0750b70453f..f6f1a7c09ce7 100755
 --- a/scripts/package/mkdebian
 +++ b/scripts/package/mkdebian
-@@ -94,10 +94,12 @@ else
+@@ -94,10 +94,13 @@ else
  	packageversion=$version-$revision
  fi
  sourcename=$KDEB_SOURCENAME
 -packagename=linux-image-$version
 -kernel_headers_packagename=linux-headers-$version
 +packagename=linux-image-$BRANCH$LOCALVERSION
 +kernel_headers_packagename=linux-headers-$BRANCH$LOCALVERSION
++libc_headers_packagename=linux-libc-dev-$BRANCH$LOCALVERSION
 +dtb_packagename=linux-dtb-$BRANCH$LOCALVERSION
  dbg_packagename=$packagename-dbg
  debarch=
 +image_name=
  set_debarch
  
  if [ "$ARCH" = "um" ] ; then
-@@ -185,6 +187,7 @@ Description: Linux kernel, version $version
+@@ -185,12 +188,13 @@ Description: Linux kernel, version $version
  
  Package: $kernel_headers_packagename
  Architecture: $debarch
 +Depends: make, gcc, libc6-dev, bison, flex, libssl-dev
  Description: Linux kernel headers for $version on $debarch
   This package provides kernel header files for $version on $debarch
   .
-@@ -205,6 +208,11 @@ Architecture: $debarch
+  This is useful for people who need to build external modules
+ 
+-Package: linux-libc-dev
++Package: $libc_headers_packagename
+ Section: devel
+ Provides: linux-kernel-headers
```

Comment:
why provides linux-kernel-header, should be linux-libc-dev?

Reply:
why provides linux-kernel-header, should be linux-libc-dev?

In the previous discussion it was decided to divide the two parts: 
1 fix creation of package 'kernel-headers' 
2 providing the 'linux-libc-dev' package

```suggestion
+#UNCOMENT	create_package "$libc_headers_packagename" "$libc_headers_dir"
```
If you need a package, uncomment the line

Reply:
my mistake, original line is provides linux-kernel-headers. do you installed it? does it conflict with debian's linux-libc-dev?

Reply:
please also add linux-libc-dev to its provide list.

Reply:
Good health. We will move in small steps. Let's fix the errors first.

Reply:
I will ask you to look carefully at the line:
+Depends: make, gcc, libc6-dev, bison, flex, libssl-dev

In any case, this has nothing to do with real 'PR'